pkg/connector/org_role.go

```diff
@@ -0,0 +1,400 @@
+package connector
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	v2 "github.com/conductorone/baton-sdk/pb/c1/connector/v2"
+	"github.com/conductorone/baton-sdk/pkg/annotations"
+	"github.com/conductorone/baton-sdk/pkg/pagination"
+	"github.com/conductorone/baton-sdk/pkg/types/entitlement"
+	"github.com/conductorone/baton-sdk/pkg/types/grant"
+	"github.com/conductorone/baton-sdk/pkg/types/resource"
+	"github.com/google/go-github/v63/github"
+	"github.com/grpc-ecosystem/go-grpc-middleware/logging/zap/ctxzap"
+	"go.uber.org/zap"
+)
+
+type OrganizationRole struct {
+	ID          int64  `json:"id"`
+	Name        string `json:"name"`
+	Description string `json:"description"`
+}
+
+type OrganizationRoleResponse struct {
+	TotalCount int                `json:"total_count"`
+	Roles      []OrganizationRole `json:"roles"`
+}
+
+type OrganizationRoleTeam struct {
+	ID   int64  `json:"id"`
+	Name string `json:"name"`
+}
+
+type orgRoleResourceType struct {
+	resourceType *v2.ResourceType
+	client       *github.Client
+	orgCache     *orgNameCache
+}
+
+func orgRoleResource(
+	ctx context.Context,
+	role *OrganizationRole,
+	org *v2.Resource,
+) (*v2.Resource, error) {
+	profile := map[string]interface{}{
+		"description": role.Description,
+	}
+
+	return resource.NewRoleResource(
+		role.Name,
+		resourceTypeOrgRole,
+		role.ID,
+		[]resource.RoleTraitOption{
+			resource.WithRoleProfile(profile),
+		},
+		resource.WithParentResourceID(org.Id),
+		resource.WithAnnotation(
+			&v2.V1Identifier{Id: fmt.Sprintf("org_role:%d", role.ID)},
+		),
+	)
+}
+
+func (o *orgRoleResourceType) ResourceType(_ context.Context) *v2.ResourceType {
+	return o.resourceType
+}
+
+func (o *orgRoleResourceType) List(
+	ctx context.Context,
+	parentID *v2.ResourceId,
+	pToken *pagination.Token,
+) ([]*v2.Resource, string, annotations.Annotations, error) {
+	if parentID == nil {
+		return nil, "", nil, nil
+	}
+
+	bag, _, err := parsePageToken(pToken.Token, &v2.ResourceId{ResourceType: resourceTypeOrgRole.Id})
+	if err != nil {
+		return nil, "", nil, err
+	}
```

Comment:
I think you can remove this and always return an empty next token if `ListRoles` is not paginated

Reply:
removed